app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -319,26 +418,40 @@ fun MapScreen(
  */
 fun drawRectangles(
     polygonAnnotationManager: PolygonAnnotationManager?,
+    plabelAnnotationManager: PointAnnotationManager?,
     locationList: List<Location>
 ) {
-  polygonAnnotationManager?.deleteAll()
+  val annotations: MutableList<PolygonAnnotationOptions> = mutableListOf()
   locationList.map { location ->
     val topLeft = location.topLeft
     val topRight = location.topRight
     val bottomLeft = location.bottomLeft
     val bottomRight = location.bottomRight
     if (topLeft != null && topRight != null && bottomLeft != null && bottomRight != null) {
-      val polygon =
-          Polygon.fromLngLats(listOf(listOf(topLeft, topRight, bottomRight, bottomLeft, topLeft)))
+      val polygon = location.toPolygon()
+
       val polygonAnnotationOptions =
           PolygonAnnotationOptions()
               .withGeometry(polygon)
-              .withFillColor("#22799B")
+              .withFillColor("#1A4988")
               .withFillOpacity(0.7)
-      polygonAnnotationManager?.create(polygonAnnotationOptions)
+      annotations.add(polygonAnnotationOptions)
+      val area = TurfMeasurement.area(polygon)
+      Log.d("MapScreen", "Area: $area")
```

Comment:
I think you forgot to remove the log here

Reply:
Fixed